atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/SnapshotStoreImpl.java

```diff
@@ -75,4 +84,18 @@ public void reset() {
     public Optional<ValueCacheSnapshot> getSnapshotForSequence(Sequence sequence) {
         return Optional.ofNullable(snapshotMap.get(sequence));
     }
+
+    private void validateStateSize() {
+        if (snapshotMap.size() > MAXIMUM_SIZE
+                || liveSequences.size() > MAXIMUM_SIZE
+                || timestampMap.size() > MAXIMUM_SIZE) {
+            log.warn(
+                    "Snapshot store has exceeded its maximum size. This likely indicates a memory leak.",
```

Comment:
Again, maybe log the minimum start timestamp in timestamp map?

Reply:
And neither are these